### ISSUE TITLE
[FR] Tracing genai metrics

### ISSUE THREAD

**Issue by Tiemen-Vanderstraeten** [mlflow/mlflow#13387]
### Willingness to contribute

Yes. I can contribute this feature independently.

### Proposal Summary

The possibility to trace the execution of the genai metrics when using mlflow.evaluate()

### Motivation

> #### What is the use case for this feature?
Track the costs of the LLM-as-a-judge metrics to get an idea of the total cost of the evaluation of a genai-application.
> #### Why is this use case valuable to support for MLflow users in general?
Costs can add up quickly if you run multiple tests on a static test set to determine the best setup, for example.
> #### Why is this use case valuable to support for your project(s) or organization?
I want to get an idea of how expensive the LLM-as-a-judge metrics are 
> #### Why is it currently difficult to achieve this use case?
I can only see it in the OpenAI platform, but not very accurately because the api is used in several parts of the application

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [X] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

**Comment by B-Step62** [comment]
Yes, we are planning to enhance tracing to genai metrics. It is a part of broader design in progress, so stay tuned!

**Comment by B-Step62** [comment]
@Tiemen-Vanderstraeten This feature is available in MLflow 3.2 (RC), please check it out🙂
https://mlflow.org/docs/latest/genai/tracing/concepts/feedback/
